src/components/ResourceList/components/BulkActions/BulkActions.scss

```diff
@@ -79,14 +62,18 @@ $bulk-actions-offset-slide-in-start: rem(-40px);
   }
 }
 
-.ButtonGroup {
-  display: flex;
-  align-items: center;
+.ButtonGroupWrapper {
   width: 100%;
-  flex-wrap: nowrap;
-  justify-content: flex-end;
-  box-shadow: inset 0 0 0 1px color('sky', 'dark');
-  border-radius: border-radius();
+  max-width: 100%;
+
+  // We need the first item of button group on small screen to fill the space
+  // stylelint-disable selector-max-specificity, selector-max-class
+  @include breakpoint-before(resource-list(breakpoint-small)) {
```

Comment:
Our `fullWidth` makes every button equal width. On a small screen, bulk actions require only the checkable button to extend the full width.

We toyed with the idea of adding a prop to ButtonGroup which was confusing and this being the only use case where we have seen this we decided to move forward with this approach. We added a [test](https://github.com/Shopify/polaris-react/pull/2467/files#diff-8b5770cb7be0cce0e7ab963b01688332R384) to ensure that if ever there's mark-up change, this will be flagged.

This is the trade-off of being able the use `ButtonGroup` and allowing us to remove quite a bit of custom components.